StyleCop.Analyzers/StyleCopTester/Program.cs

```diff
@@ -59,7 +65,20 @@ private static IEnumerable<DiagnosticAnalyzer> FilterAnalyzers(IEnumerable<Diagn
 
             foreach (var analyzer in analyzers)
             {
-                if (useAll || analyzer.SupportedDiagnostics.Any(y => ids.Contains(y.Id)))
+                if (useAll)
+                {
+                    yield return analyzer;
+                }
```

Comment:
It wouldn't really make a difference. This is not performance-critical code (it's a one-time initialization in a test application). Also, it would still need to use a `foreach` loop because of the `yield return` statements.